database/mongo/controller.mjs

```diff
@@ -0,0 +1,237 @@
+            return err
+        }
+    }
+
+    /** 
+     * Generally check that the controller has an active connection 
+     * @return boolean
+     * */
+    async connected() {
+        // Send a ping to confirm a successful connection
+        try {
+            let result = await this.db.command({ ping: 1 }).catch(err => { return false })
+            result = result.ok ? true : false
+            return result
+        } catch (err) {
+            console.error(err)
+            return false
+        }
+    }
+
+    /**
+     * Get by property matches and return all objects that match
+     * @param query JSON from an HTTP POST request.  It must contain at least one property.
+     * @return JSON Array of matched documents or standard error object
+     */
+    async read(query) {
+        try {
+            //need to determine what collection (projects, groups, userPerferences) this goes into.
+            const data_type = query["@type"] ?? query.type ?? null
+            if (!data_type)
+                return { "endpoint_error": "find", "status": 400, "message": `Cannot find 'type' on this data, and so cannot figure out a collection for it.` }
+            const collection = discernCollectionFromType(data_type)
+            if (!collection)
+                return { "endpoint_error": "find", "status": 400, "message": `Cannot figure which collection for object of type '${data_type}'` }
+            if (Object.keys(query).length === 0)
+                return { "endpoint_error": "find", "status": 400, "message": `Empty or null object detected.  You must provide a query object.` }
+            let result = await this.db.collection(collection).find(query).toArray()
+            return result
+        } catch (err) {
+            console.error(err)
+            return { "endpoint_error": "find", "status": 500, "message": "There was an error querying the database." }
+        }
+    }
+
+    /**
+     * Insert a document into the database (mongo)
+     * @param data JSON from an HTTP POST request
+     * @return The inserted document JSON or error JSON
+     */
+    async create(data) {
+        try {
+            //need to determine what collection (projects, groups, userPerferences) this goes into.
+            const data_type = data["@type"] ?? data.type ?? null
+            if (!data_type)
+                return { "endpoint_error": "insertOne", "status": 400, "message": `Cannot find 'type' on this data, and so cannot figure out a collection for it.` }
+            const collection = discernCollectionFromType(data_type)
+            if (!collection)
+                return { "endpoint_error": "insertOne", "status": 400, "message": `Cannot figure which collection for object of type '${data_type}'` }
+            const id = this.newID()
+            data["_id"] = id
+            const result = await this.db.collection(collection).insertOne(data)
+            if (result.insertedId) {
+                return data
+            } else {
```

Comment:
These `else`s are removable if you just flip them all. Not wrong, but might be easier to read, since the rest of the function is just kicking away errors and that would leave the default successful behavior as a data return.